File: 51-60/Q56.py
#Q56.py
class Solution:
    def merge(self, a):
        a.sort()
        res=[]
        for s,e in a:
            if not res or res[-1][1] < s:
                res.append([s,e])
            else:
                res[-1][1] = max(res[-1][1], e)
        return res

File: 51-60/Q56.cpp
//Q56.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& a) {
        sort(a.begin(), a.end());
        vector<vector<int>> res;
        for(auto &v:a){
            if(res.empty() || res.back()[1] < v[0])
                res.push_back(v);
            else
                res.back()[1] = max(res.back()[1], v[1]);
        }
        return res;
    }
};
